src/broadcast/participant.rs

```diff
@@ -94,6 +105,14 @@ impl ProtocolParticipant for BroadcastParticipant {
     ) -> Result<ProcessOutcome<Self::Output>> {
         info!("Processing broadcast message.");
 
+        // XXX Protocols that utilize `BroadcastParticipant` never "reset" the
```

Comment:
This is still an open question: how should we handle protocol termination and restart in `BroadcastParticipant`?

Reply:
Hmm. None of our subprotocols have more than one broadcast round, though. With the new single-use-participant architecture, I think this won't be a problem. Were you seeing it pop up in tests?

It does raise a question of whether we want to handle instantiation of the broadcast participant differently, because the current structure does imply reusability.

Reply:
Yeah, a bunch of the tests fail with this included. Let me try to trace down exactly why.